draft-thomson-http-mice.md

```diff
@@ -264,62 +266,50 @@ status code.  However, if the integrity proof for the first record is not known,
 this check SHOULD NOT fail unless explicitly configured to do so.
 
 
-# The MI HTTP Header Field {#header}
+# The mi-sha256 digest algorithm {#digest-mi-sha256}
 
-The MI HTTP header field carries message integrity proofs corresponding to
-content encoding(s) that have been applied to a payload body.
+[RFC3230] describes digests applying to "the entire instance associated with the
+message". The instance corresponds to the "representation" in Section 3 of
+[RFC7231], but unlike the existing digest algorithms, the mi-sha256 digest
+algorithm specifies the top-level digest at the point when the mi-sha256 content
+coding ({{encoding}}) is applied or removed from the representation.
 
-The MI header field uses the extended ABNF syntax defined in Section 1.2 of
-[RFC7230] and the `parameter` rule from [RFC7231]:
+When the "mi-sha256" digest algorithm is specified for a representation, the
+recipient MUST:
 
-~~~
-  MI = #mi_params
-  mi_params = [ parameter *( ";" parameter ) ]
-~~~
-
-If the payload is encoded more than once (as reflected by having multiple
-content-codings that use the message integrity header field), each application
-of the content encoding is reflected in the MI header field in the order in
-which they were applied.
-
-The MI header MAY be omitted if the sender intends for the receiver to acquire
-the integrity proof for the first record by other means.
-
-*RFC EDITOR: Please remove the next paragraph before publication.*
-
-Implementations of drafts of this specification use the "MI" header field
-instead of renaming it to MI-## like the "mi-sha256" content encoding.
-
-
-## MI Header Field Parameters
-
-The following parameters are used in validating content encoded with the
-"mi-sha256" content encoding:
+1. If the "mi-sha256" content coding has not been applied to the representation
+   exactly once (Section 3.1.2.2 of [RFC7231]), reject the representation.
```

Comment:
This is a change.  Are you concerned that Digest has rules that would prevent this content coding from being used twice on the same representation?

Reply:
More assumptions than explicit rules: although [Digest](https://tools.ietf.org/html/rfc3230#section-4.3.2) does hold a list of instance-digests, and does say that it "MAY contain multiple instance-digest values", it doesn't specify that the order is significant, and occasionally implies that a particular digest-algorithm only produces one digest ("The digest is computed on the entire instance associated with the message.")

We could override this and specify explicitly which of several repeated Digest entries applies to which repeated mi-sha256 coding, as the MI header did, but it didn't seem particularly useful to repeat mi-sha256 encodings, so I didn't worry about it. I can do it if you prefer.

It will, of course, be useful to apply an mi-sha256 encoding followed by an mi-some-new-hash encoding, but this wording was meant to allow that.